src/main/java/com/o19s/es/ltr/query/RankerQuery.java

```diff
@@ -360,11 +363,11 @@ private void advanceSubIterators(int target) throws IOException {
             if (target == NO_MORE_DOCS) {
                 return;
             }
-            for (DocIdSetIterator iterator : subIterators) {
-                // FIXME: Probably inefficient
-                if (iterator.docID() < target) {
-                    iterator.advance(target);
-                }
+            while (subIteratorsPriorityQueue.top().iterator.docID() < target) {
+                DisiWrapper disiWrapper = subIteratorsPriorityQueue.pop();
```

Comment:
should we call updateTop() after calling advance instead of removing it?

Reply:
good point! fixed